### PR TITLE
CRN-873 events search

### PR DESCRIPTION
**Pull request by niezgoda**

### CODE REVIEW

File: apps/crn-server/src/handlers/event/index-user-events-handler.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { algoliaApiKey, algoliaAppId, algoliaIndex } from '../../config';
+import Events, { EventController } from '../../controllers/events';
+import logger from '../../utils/logger';
+import {
+  loopOverCustomCollection,
+  LoopOverCustomCollectionFetchOptions,
+} from '../../utils/loop-over-custom-colection';
+import { UserEvent, UserPayload } from '../event-bus';
+
+export const indexUserEventsHandler =
```
Comment (markcsims) [inline apps/crn-server/src/handlers/event/index-user-events-handler.ts:17]: All these handlers seem to be doing the same thing (with different filters).

Is there a way we can refactor to share the code across them?

**Review by markcsims — CHANGES_REQUESTED** [review]

File: apps/crn-frontend/src/events/state.ts [hunk 2]
```diff
@@ -88,18 +90,36 @@ export const useQuietRefreshEventById = (id: string) => {
...
+  const isEventsSearchFromAlgoliaEnabled =
+    useFlags().isEnabled('EVENTS_SEARCH');
   const [events, setEvents] = useRecoilState(eventsState(options));
   useDeepCompareEffect(() => {
     if (events === undefined) {
-      getEvents(options, authorization).then(setEvents).catch();
+      if (isEventsSearchFromAlgoliaEnabled) {
+        getEventsFromAlgolia(algoliaClient.client, options)
+          .then(setEvents)
+          .catch();
```
[the remark this answers is not in the document]
Comment (niezgoda) [reply apps/crn-frontend/src/events/state.ts:102]: it is the same functionality as previously

File: apps/crn-frontend/src/events/api.ts [hunk 3]
```diff
@@ -1,10 +1,35 @@
...
+  algoliaClient: AlgoliaSearchClient,
+  { searchQuery, currentPage, pageSize, before, after }: GetEventListOptions,
+): Promise<ListEventResponse> => {
+  const algoliaFilters: string[] = [];
+
+  if (before) {
+    const beforeTimestamp = Math.round(new Date(before).getTime() / 1000);
+    algoliaFilters.push(`endDateTimestamp < ${beforeTimestamp}`);
+  } else if (after) {
+    const afterTimestamp = Math.round(new Date(after).getTime() / 1000);
```
[the remark this answers is not in the document]
Comment (niezgoda) [reply apps/crn-frontend/src/events/api.ts:19]: yes. but the condition is different

File: apps/crn-server/src/handlers/event/index-user-events-handler.ts (hunk 1 above)
Comment (niezgoda) [reply apps/crn-server/src/handlers/event/index-user-events-handler.ts:17]: I do not think we want to do this

**Review by markcsims — APPROVED** [review]

**Pull request merged**